node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/NodePatcher.java

```diff
@@ -58,35 +66,37 @@ public NodePatcher(NodeFlavors nodeFlavors, InputStream json, Node node, NodeRep
         inspector.traverse((String name, Inspector value) -> {
             try {
                 node = applyField(name, value);
-            }
-            catch (IllegalArgumentException e) {
+            } catch (IllegalArgumentException e) {
                 throw new IllegalArgumentException("Could not set field '" + name + "'", e);
             }
-        } );
 
+            try {
+                children = applyFieldRecursive(children, name, value);
+                childrenModified = true;
+            } catch (IllegalArgumentException e) {
+                // Non recursive field, ignore
+            }
+        } );
 
-        List<Node> nodes = new ArrayList<>();
-        if (node.type() == NodeType.host) {
-            nodes.addAll(modifiedDockerChildNodes());
-        }
+        List<Node> nodes = childrenModified ? new ArrayList<>(children) : new ArrayList<>();
         nodes.add(node);
 
         return nodes;
     }
 
-    private List<Node> modifiedDockerChildNodes() {
-        List<Node> children = nodeRepository.getChildNodes(node.hostname());
-        boolean modified = false;
-
-        if (inspector.field(HARDWARE_FAILURE_DESCRIPTION).valid()) {
-            Optional<String> hardwareFailure = asOptionalString(inspector.field(HARDWARE_FAILURE_DESCRIPTION));
-            modified = true;
-            children = children.stream()
-                    .map(node -> node.with(node.status().withHardwareFailureDescription(hardwareFailure)))
-                    .collect(Collectors.toList());
+    private List<Node> applyFieldRecursive(List<Node> childNodes, String name, Inspector value) {
+        switch (name) {
+            case HARDWARE_FAILURE_DESCRIPTION:
+                return childNodes.stream()
+                        .map(child -> child.with(child.status().withHardwareFailureDescription(asOptionalString(value))))
+                        .collect(Collectors.toList());
+            case WANT_TO_RETIRE:
+                return childNodes.stream()
+                        .map(child -> child.with(child.status().withWantToRetire(asBoolean(value))))
+                        .collect(Collectors.toList());
+            default :
+                throw new IllegalArgumentException("Field " + name + " is not recursive");
```

Comment:
I suggest just returning `Collections.emptyList` as this exception seems to always be ignored.

Reply:
Yes, but that would require that we save the return in a temporary variable and only overwrite `nodes` if the temporary variable is not empty. I prefer it this way because at least this one has some symmetry with the `applyField()` call above.